### DISCUSSION TITLE
How to properly connect local node without internet peer?

### DISCUSSION THREAD

**Discussion by d47081** [yggdrasil-network/yggdrasil-go#1136] [Q&A]
I want to connect another machine in the local network without internet peer at middle point.
As understand,  on target node, I must provide host/port in the `Listen` section, but this comment confusing a bit:

> listeners in order to accept incoming peerings from non-local nodes

``` yggdrasil.conf
  # Listen addresses for incoming connections. You will need to add
  # listeners in order to accept incoming peerings from non-local nodes.
  # Multicast peer discovery will work regardless of any listeners set
  # here. Each listener should be specified in URI format as above, e.g.
  # tls://0.0.0.0:0 or tls://[::]:0 to listen on all interfaces.
  Listen: []
```

But there is no other way but `tls://192.x.x.x:123` in `Listen` on target, and `tls://192.x.x.x:123` as `Peers` on connecting node?
Or how?

**Comment by tribals** [comment]
For that you need to utilize multicast peers discovery (it's usage is by means limited only to local network segment). For that, you need to [configure `MulticastInterfaces` option](https://yggdrasil-network.github.io/configurationref.html#multicastinterfaces). In other words, you don't need to peer (verb.) with local peers manually - just enable multicast peering.

If you still want to employ `Peers` (note: not `Listen`) option for local peerings, then... you can do even that. To do so, you need to assign permanent addresses for peers to which you're trying to connect (again, through `Peers` option; of course, you need at least one `Listen` on the other end, like `tcp://[::]:4242`), but that is the main issue which multicast peering tries to solve - addresses on local network can change unpredictably, so you will end up updating config again and again. You still can mitigate that by some means of local addressing like mDNS, though. It is not clear why to do that if there is already multicast peering?
